src/Player.cpp

```diff
@@ -1929,15 +1929,14 @@ void Player::setQuestProgress(TYPE_OF_QUEST_ID questid, TYPE_OF_QUESTSTATUS prog
     int timeNow = int(time(nullptr));
 
     try {
-        connection->beginTransaction();
-
-        SelectQuery query(connection);
-        query.addColumn("questprogress", "qpg_progress");
-        query.addEqualCondition<TYPE_OF_CHARACTER_ID>("questprogress", "qpg_userid", getId());
-        query.addEqualCondition<TYPE_OF_QUEST_ID>("questprogress", "qpg_questid", questid);
-        query.addServerTable("questprogress");
+        UpdateQuery updQuery;
+        updQuery.addAssignColumn<TYPE_OF_QUESTSTATUS>("qpg_progress", progress);
+        updQuery.addAssignColumn<int>("qpg_time", timeNow);
+        updQuery.addEqualCondition<TYPE_OF_CHARACTER_ID>("questprogress", "qpg_userid", getId());
+        updQuery.addEqualCondition<TYPE_OF_QUEST_ID>("questprogress", "qpg_questid", questid);
+        updQuery.setServerTable("questprogress");
 
-        Result results = query.execute();
+        Result results = updQuery.execute();
```

Comment:
`results` will always be empty for updates. Below we need to check if `results.affected_rows() == 0` instead.